sdlib/d/gc/arena.d

```diff
@@ -244,6 +273,28 @@ private:
 		(cast(Arena*) &this).freePagesImpl(e, n, pages);
 	}
 
+	void shrinkAlloc(shared(ExtentMap)* emap, Extent* e, uint delta) shared {
```

Comment:
The flip flop between new size and delta makes thing a bit hard to follow - and probably hide some redundant computation.